### PR TITLE
Update tally

### PR DESCRIPTION
**Pull request by DonAurelio**
We updated the old version of tally to the metababel based version.

### CODE REVIEW

File: .github/workflows/presubmit.yml
Comment (DonAurelio) [inline .github/workflows/presubmit.yml:72]: Oh, will fix this

File: .github/workflows/presubmit.yml
Comment (DonAurelio) [inline .github/workflows/presubmit.yml:72]: Will fix this

File: xprof/btx_tally_params.yaml
Comment (DonAurelio) [inline xprof/btx_tally_params.yaml:1]: Do you think we should made the default param values explicit in tally, that would be clearer for the user?

Comment (DonAurelio) [inline xprof/btx_tally_params.yaml:1]: Do you think we should made the default param values explicit in tally, that would be clearer for the user?

File: xprof/Makefile.am
Comment (TApplencourt) [inline xprof/Makefile.am:23]: Same problem, we cannot have 2 output in make. It will break parallel build.

File: xprof/btx_tally_params.yaml
Comment (TApplencourt) [reply xprof/btx_tally_params.yaml:1]: What do you mean? No default value?

File: xprof/tally.hpp
Comment (TApplencourt) [inline xprof/tally.hpp:119]: what all those TODO mean?

File: xprof/tally.hpp
Comment (TApplencourt) [inline xprof/tally.hpp:313]: `duratiin` not english :)

File: xprof/tally.cpp
Comment (TApplencourt) [inline xprof/tally.cpp:21]: Please don't do that. Just do `data->param  = usr_params`

File: xprof/btx_tally_params.yaml
Comment (DonAurelio) [reply xprof/btx_tally_params.yaml:1]: For example: the default value of `display_compact` was put explicitly, but `demangle_name` don't. Should all param default values be explicit?

```yaml
:type: map
:entries:
  - :name: display_compact
    :type: bool
    :default_value: BT_TRUE
  - :name: demangle_name
    :type: bool
```

File: xprof/Makefile.am
Comment (DonAurelio) [reply xprof/Makefile.am:23]: Just to be sure I am getting the right message.

`btx_tally_upstream_model.yaml btx_timeline_upstream_model.yaml: btx_interval_model.yaml
`
is simular to 

`%.pb.h %.pb.cc: %.proto
`
So both will breack the parallel build ?

File: xprof/tally.cpp
Comment (DonAurelio) [reply xprof/tally.cpp:4]: Agree, we do not need it. It was becasue we were measuring tabulation time.

File: xprof/Makefile.am
Comment (TApplencourt) [reply xprof/Makefile.am:23]: What @Kerilk  is saying is that the `%.` is more robust to parallel build.

File: xprof/tally.hpp
Comment (TApplencourt) [reply xprof/tally.hpp]: What we need is grouped target (https://www.gnu.org/software/make/manual/html_node/Multiple-Targets.html) but this is a newer version of make...

Comment (DonAurelio) [reply xprof/tally.hpp]: Good advice !!

**Comment by TApplencourt** [comment]
>+1,112 −1,076
Look better :D

File: xprof/tally.hpp
Comment (TApplencourt) [inline xprof/tally.hpp:3]:
```suggestion
//! This file contains the infrastructure used to tabulate data 
```

File: xprof/Makefile.am
Comment (TApplencourt) [inline xprof/Makefile.am:62]: I think you need to remove the space (if this is a space, should be only tab)

File: xprof/tally.hpp
Comment (TApplencourt) [inline xprof/tally.hpp:115]: Do you think we need those comment? They seem pretty obvious to me

File: xprof/tally.hpp
Comment (TApplencourt) [inline xprof/tally.hpp:261]: Can we move this one in the tally.cpp?

File: xprof/tally.hpp
Comment (TApplencourt) [inline xprof/tally.hpp:457]: What is the pointer of this comment?

File: xprof/tally.hpp
Comment (DonAurelio) [reply xprof/tally.hpp:115]: We actually don't need them, they helped my to understand, now we can delete them.

File: xprof/tally.hpp
Comment (TApplencourt) [inline xprof/tally.hpp:489]: Should we remove this comment?

File: xprof/tally.hpp
Comment (TApplencourt) [inline xprof/tally.hpp:812]: I think this can be in the tally.cpp, should not be in the cpp for sure

File: xprof/btx_timeline_params.yaml
Comment (TApplencourt) [inline xprof/btx_timeline_params.yaml:1]: does it work if it's just an empty file?
Maybe we can put a comment inside to said "required because of global rule in makefile" or something like that

File: xprof/tally.hpp
Comment (TApplencourt) [reply xprof/tally.hpp:115]: We can keep the other comment; I think they help us understand the complicated C++ code. But those one I think we can delete

File: xprof/tally.hpp
Comment (DonAurelio) [reply xprof/tally.hpp:812]: What should be the criteria for considering moving this?

File: xprof/tally.hpp
Comment (DonAurelio) [reply xprof/tally.hpp:457]: No point, some unused function i wasn't sure to delete. Now it is gone !!!

File: xprof/tally.hpp
Comment (TApplencourt) [reply xprof/tally.hpp]: it's function non-templated. The definition/implementation have nothing to do in the `.hpp`.
So you can put the declaration inside the `hpp` and then the implementation in the `cpp`.
But this function is only used on the `cpp`, so no need to export the symbol

File: xprof/tally.cpp
Comment (TApplencourt) [inline xprof/tally.cpp:40]: I think we can put the first `unsigned` correspond to the `level` (it's non obvious, maybe we should put an ew type to make it clearer) and then merge all the comments. It's obvious what traffic, device, host mean IMO

File: xprof/tally.cpp
Comment (TApplencourt) [inline xprof/tally.cpp:97]: Please run `clang-format  -style="{ColumnLimit: 100}"`  (module load llvm on JLSE)

**Review by TApplencourt — APPROVED** [review]

**Comment by TApplencourt** [comment]
LGTM! Let wait for @Kerilk  review before merging.

File: xprof/Makefile.am
Comment (Kerilk) [inline xprof/Makefile.am:25]: Split into two commands to avoid using recent group target feature until really required.

File: utils/babeltrace_thapi.in
Comment (Kerilk) [inline utils/babeltrace_thapi.in:179]: Maybe try to gremove the btx_ prefix from the component class.

File: xprof/btx_interval_model.yaml
Comment (Kerilk) [inline xprof/btx_interval_model.yaml:1]: :stream_classes:
  - :name: some_name

Maybe use a more meaningful name.

**Review by Kerilk — CHANGES_REQUESTED** [review]
A few comments.

File: xprof/btx_interval_model.yaml
Comment (TApplencourt) [inline xprof/btx_interval_model.yaml]: Remove the space

File: xprof/xprof.sh.erb.in
Comment (TApplencourt) [inline xprof/xprof.sh.erb.in:455]: Why the default changed for `display_metadata=`?

File: xprof/xprof.sh.erb.in
Comment (TApplencourt) [inline xprof/xprof.sh.erb.in:454]: Why did we switch to a bool for this one?
I think it make sense to have `text/human` or `json` ?

File: xprof/xprof.sh.erb.in
Comment (TApplencourt) [inline xprof/xprof.sh.erb.in:452]: Also why changing for this one?
We may have more mode ? Like for MPI we may want to have only `min` / `max` rank

File: xprof/xprof.sh.erb.in
Comment (TApplencourt) [inline xprof/xprof.sh.erb.in:1]: If we need to add the support for string in params of metababel we need to do it

Comment (DonAurelio) [reply xprof/xprof.sh.erb.in:1]: Although `display` is pased as a string here, it ends up becoming a boolean, as shown [here](https://github.com/argonne-lcf/THAPI/blob/9a96ead21c1b25145f24e6f955d9a85acc4726be/xprof/tally.cpp#LL222C2-L222C2) . I think the bolean is cleaner. But of couse, once we have more printing modes we can change it to string, but when we need it. What do you think?

Comment (DonAurelio) [reply xprof/xprof.sh.erb.in:1]: Nevermind, I have to put it as It was !!!

**Review by TApplencourt — APPROVED** [review]

**Comment by TApplencourt** [comment]
🎉

**Pull request merged**
